### DISCUSSION TITLE
RTK Query: Best way to fetch two endpoints (one after the other) ?

### DISCUSSION THREAD

**Discussion by AlbinCederblad** [reduxjs/redux-toolkit#1367] [Q&A] [answered]
I need to query the first endpoint to get an ID to be able to fetch the correct data from the 2nd endpoint.

Any way to do this with RTK Query? I'm trying to convert as much as possible to use RTK Query instead of action/reducer/slice.

Thanks!

**Comment by AlbinCederblad** [comment, marked as answer]
i think i found it; https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#performing-multiple-requests-with-a-single-query

**Reply by phryneas** [reply]
Either that or just using the two `useSomeQuery` hooks directly after the other, the second one skipping until the first one is complete.

**Reply by AlbinCederblad** [reply]
Ok, thanks! That sounds easier. Is there any example of doing that?

**Reply by phryneas** [reply]
I'm sure there was but can't find it right now.

```js
const result = useSomeQuery(arg)
const result2 = useSomeOtherQuery(result.sucess ? arg : skipToken)
```
